### PR TITLE
Correcting "New font icons, Modernizr, and A Font Garde"

### PR DESCRIPTION
**Pull request by talbs**
This work revises and builds upon the WIP PR https://github.com/edx/ux-pattern-library/pull/173 to make the conversion to the new icon method near complete.

- - -

##### TODOs

* [ ] Fix the rest of the script linting warnings found with the new ``modernizr.fontface-generatedcontent.js`` vendor file
* [ ] along side https://github.com/edx/ux-pattern-library/commit/7662981f6c2288aa3b8875b4cf86a7a686390124, address revising the syntax of icons used in enhanced form select/dropdown fields.

### CODE REVIEW

**Comment by talbs** [comment]
@clrux, please review and let me know when this can be merged into your work to get this very close to done. Also, see the TODOs for once it is merged. I'll need you to tackle the rest of those as they're JS-based with some things you've written.

FYI, @frrrances

**Comment by talbs** [comment]
@clrux, also note all of my commits with the ``fixup!`` prefix. Once merged, please try to squash them into any commits that are already starting related work or rename them to be standalone commits for this repo's history.

File: pattern-library/sass/_edx-pattern-library.scss [hunk 1]
```diff
@@ -21,6 +21,7 @@
 @import 'utilities/normalize';
 @import 'utilities/reset';
 @import 'utilities/typography';
+@import 'utilities/afontgarde';
```
Comment (clrux) [inline pattern-library/sass/_edx-pattern-library.scss]: Does this work for you given it's a CSS file?

File: pattern-library/sass/utilities/_afontgarde.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/*! afontgarde - v0.1.6 - 2015-03-13
```
Comment (clrux) [inline pattern-library/sass/utilities/_afontgarde.scss]: Nevermind, you made it a Sass partial. Kewl.

**Comment by clrux** [comment]
It looks like you've taken care of nearly all of the PR feedback from my PR, so thanks. This looks good.

:+1:

File: pattern-library/sass/utilities/_afontgarde.scss (hunk 2 above)
Comment (talbs) [inline pattern-library/sass/utilities/_afontgarde.scss]: Yeah, I did the same with ``_syntax.scss`` borrowed code used for code syntax styling in the Doc Site. Its not worth keeping things as ``.css`` files in a static vendor dir if they're not a bower package IMHO. I'll note in the README that A Font Garde is another tool we're using.

Comment (talbs) [inline pattern-library/sass/utilities/_afontgarde.scss]: I commented in a known Sass github issue around being able to import plain old CSS files in a Sass import (https://github.com/sass/sass/issues/279#issuecomment-136926782), FYI.

**Comment by talbs** [comment]
Thanks much for the response, @clrux. I'll merge and you can wrap up loose ends while @frrrances and @cptvitamin have a once over to be cool with firming up this really foundational work.

**Comment by clrux** [comment]
@talbs I've fixed the second of the to-dos, but... with the first (regarding Modernizr), I'm hesitant to fix the warnings for two reasons:

* it works as it is
* it's stock as it is

Annoying as they may be... what do you think, sir?

**Comment by talbs** [comment]
@clrux, thanks for the consideration and mindfulness. I'd still lean on making these syntax fixes as:

* We're already not treating this as a full blown vendor package (not using bower or a vendor dir)
* We're less likely to have to update this solution in the future than other libraries
* This version of modernizr is already customized by Filament Group
* If we don't correct these issues, anyone using the UXPL gulp workflow will be presented with linting errors/warnings every time they compile assets (and this could happen in other places using the UXPL bower package that also use linting on their scripts).
* I've already "beautified" the vendor JS to expand it and have made the syntax changes I could to reduce lint warnings

**With that mind, can you fix the syntax so that no lint warnings appear during the ``gulp`` command?**

If you still feel queasy about it (which I can understand), the real right thing to do is to 1) fix the syntax so no warnings appear, 2) re-uglify the file, and 3) make a PR against A Font Garde to get these syntax fixes into their repo. I'm all for that and even if you don't want to or I can't, I probably will.

**Pull request merged**